code/include/playfab_c/PlayFabClientDataModels_c.h

```diff
@@ -0,0 +1,495 @@
+{
+    ApplePushNotificationService = 0,
+    GoogleCloudMessaging = 1
+};
+
+struct PlayFabPlayerProfileViewConstraints
+{
+    bool showAvatarUrl;
+    bool showBannedUntil;
+    bool showCampaignAttributions;
+    bool showContactEmailAddresses;
+    bool showCreated;
+    bool showDisplayName;
+    bool showExperimentVariants;
+    bool showLastLogin;
+    bool showLinkedAccounts;
+    bool showLocations;
+    bool showMemberships;
+    bool showOrigination;
+    bool showPushNotificationRegistrations;
+    bool showStatistics;
+    bool showTags;
+    bool showTotalValueToDateInUsd;
+    bool showValuesToDate;
+};
+
+struct PlayFabGetPlayerCombinedInfoRequestParams
+{
+    bool getCharacterInventories;
+    bool getCharacterList;
+    bool getPlayerProfile;
+    bool getPlayerStatistics;
+    bool getTitleData;
+    bool getUserAccountInfo;
+    bool getUserData;
+    bool getUserInventory;
+    bool getUserReadOnlyData;
+    bool getUserVirtualCurrency;
+    const char** playerStatisticNames;
+    size_t playerStatisticNamesCount;
+    PlayFabPlayerProfileViewConstraints* profileConstraints;
+    const char** titleDataKeys;
+    size_t titleDataKeysCount;
+    const char** userDataKeys;
+    size_t userDataKeysCount;
+    const char** userReadOnlyDataKeys;
+    size_t userReadOnlyDataKeysCount;
+};
+
+struct PlayFabLoginWithCustomIDRequest
+{
+    bool* createAccount;
+    const char* customId;
+    PlayFabDictionaryEntry* customTags;
+    size_t customTagsCount;
+    const char* encryptedRequest;
+    PlayFabGetPlayerCombinedInfoRequestParams* infoRequestParameters;
+    const char* playerSecret;
+    const char* titleId;
+};
+
+struct PlayFabAdCampaignAttributionModel
+{
+    time_t attributedAt;
+    const char* campaignId;
+    const char* platform;
+};
+
+struct PlayFabContactEmailInfoModel
+{
+    const char* emailAddress;
+    const char* name;
+    PlayFabEmailVerificationStatus* verificationStatus;
+};
+
+struct PlayFabLinkedPlatformAccountModel
+{
+    const char* email;
+    PlayFabLoginIdentityProvider* platform;
+    const char* platformUserId;
+    const char* username;
+};
+
+struct PlayFabLocationModel
+{
+    const char* city;
+    PlayFabContinentCode* pfContinentCode;
+    PlayFabCountryCode* pfCountryCode;
+    double* latitude;
+    double* longitude;
+};
+
+struct PlayFabSubscriptionModel
+{
+    time_t expiration;
+    time_t initialSubscriptionTime;
+    bool isActive;
+    PlayFabSubscriptionProviderStatus* status;
+    const char* subscriptionId;
+    const char* subscriptionItemId;
+    const char* subscriptionProvider;
+};
+
+struct PlayFabMembershipModel
+{
+    bool isActive;
+    time_t membershipExpiration;
+    const char* membershipId;
+    time_t* overrideExpiration;
+    PlayFabSubscriptionModel** subscriptions;
+    size_t subscriptionsCount;
+};
+
+struct PlayFabPushNotificationRegistrationModel
+{
+    const char* notificationEndpointARN;
+    PlayFabPushNotificationPlatform* platform;
+};

```

Comment:
should be ptr?

Reply:
see above

---
In reply to: [590792111](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r590792111) [](ancestors = 590792111)